goaviatrix/account.go

```diff
@@ -218,18 +219,38 @@ func (c *Client) CreateAWSSAccount(account *Account) error {
 	return c.PostFileAPI(params, files, DuplicateBasicCheck)
 }
 
-func (c *Client) GetAccount(account *Account) (*Account, error) {
+func (c *Client) ListAccounts() ([]Account, error) {
+	// If cached accounts are recent enough, return them
+	c.cacheMutex.Lock()
```

Comment:
Do we need to invalidate the cache when an account is added, updated or delete?

Reply:
Each command is stateless, so for instance, if I add a new account to my TF config, it should only see that diff.  This function should only affect read back.  I assume in this case you mean account in the terraform config right?

Lets say I have a bunch of accounts already defined and I add a new one, I should only see this:

```
aviatrix_account.aws_account[410]: Refreshing state... [id=aws-411]
aviatrix_account.aws_account[4]: Refreshing state... [id=aws-5]
aviatrix_account.aws_account[324]: Refreshing state... [id=aws-325]
aviatrix_account.aws_account[46]: Refreshing state... [id=aws-47]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aviatrix_account.azure_account-2 will be created
  + resource "aviatrix_account" "azure_account-2" {
      + account_name            = "azure-new-account"
      + arm_application_id      = (sensitive value)
      + arm_application_key     = (sensitive value)
      + arm_directory_id        = (sensitive value)
      + arm_subscription_id     = (sensitive value)
      + audit_account           = false
      + aws_ca_cert_path        = (known after apply)
      + aws_role_app            = (known after apply)
      + aws_role_ec2            = (known after apply)
      + awschina_role_app       = (known after apply)
      + awschina_role_ec2       = (known after apply)
      + awsgov_role_app         = (known after apply)
      + awsgov_role_ec2         = (known after apply)
      + awss_cap_cert_key_path  = (known after apply)
      + awss_cap_cert_path      = (known after apply)
      + awsts_cap_cert_key_path = (known after apply)
      + awsts_cap_cert_path     = (known after apply)
      + cloud_type              = 8
      + id                      = (known after apply)
    }

Plan: 1 to add, 0 to change, 0 to destroy.
````

Same for if I delete one.

Does that answer the question?

Reply:
I have to admit I didn't look at the Terraform's component model yet.
I was thinking in terms of 
1. Terraform gets the AWS account to reconcile resource A
2. Terraform creates updates the AWS account
3. Terraform stores the AWS account in the output and finds the original account in the cache.

But yeah, I am not really familiar with the component model, so I am probably wrong.

Reply:
Its a good point, let me convince myself that can't happen.